modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -490,16 +491,22 @@ object ObservationService {
           raEnd: RightAscension,
           decStart: Declination,
           decEnd: Declination,
-          active: TimestampInterval
+          site: Site,
+          active: DateInterval
         ): F[Option[ObservationValidation]] = 
           asterismService.getAsterism(pid,oid)
             .map { l =>
               val targets = l.map(_._2)
               // The lack of a target will get reported in the generator errors
+              val start    = ObservingNight.fromSiteAndLocalDate(site, active.start).start
+              val end      = ObservingNight.fromSiteAndLocalDate(site, active.end).end
+              val duration = Duration.between(start, end)
+              val center   = start.plus(duration.dividedBy(2L))
```

Comment:
Are we calculating the position at the middle of the night? Is this just for some validations?

Reply:
Yeah this is the conversion of the old absolute `TimestampInterval` active period check to use a `DateInterval` applied to a specific site.